core/src/CMakeLists.txt

```diff
@@ -68,12 +75,24 @@ IF (KOKKOS_ENABLE_SYCL)
   APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/SYCL/*.hpp)
 ENDIF()
 
-IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS)
-  APPEND_GLOB(KOKKOS_CORE_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/desul/src/*.cpp)
-  APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/desul/*.hpp)
-  APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/desul/*/*.hpp)
-  APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/desul/*/*/*.hpp)
-  APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/desul/*/*/*.inc)
+IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS AND NOT desul_atomics_FOUND)
+  APPEND_GLOB(KOKKOS_CORE_SRCS ${CMAKE_SOURCE_DIR}/tpls/desul/src/*.cpp)
+  APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_SOURCE_DIR}/tpls/desul/include/desul/*.hpp)
+  APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_SOURCE_DIR}/tpls/desul/include/desul/*/*.hpp)
+  APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_SOURCE_DIR}/tpls/desul/include/desul/*/*/*.hpp)
+  APPEND_GLOB(KOKKOS_CORE_HEADERS ${CMAKE_SOURCE_DIR}/tpls/desul/include/*/*/*.inc*)
+
+  INSTALL (DIRECTORY
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../tpls/desul/include/desul"
+    DESTINATION ${KOKKOS_HEADER_DIR}
+    FILES_MATCHING
+    PATTERN "*.inc"
+    PATTERN "*.inc_*"
+    PATTERN "*.hpp"
+  )
+  MESSAGE(STATUS "Using internal DESUL atomics copy")
+ELSE()
+  MESSAGE(STATUS "Using external DESUL atomics copy")
```

Comment:
Print the path to the package that was found.

Reply:
how?